Tests/BugsnagEventTests.m

```diff
@@ -539,6 +540,47 @@ - (void)testApiKey {
     }];
 }
 
+- (void)testStacktraceTypes {
+    BugsnagEvent *event = [[BugsnagEvent alloc] init];
+    XCTAssertEqualObjects(event.stacktraceTypes, @[]);
+    
+    BugsnagError *error = [[BugsnagError alloc] init];
+    event.errors = @[error];
+    error.type = BSGErrorTypeCocoa;
+    XCTAssertEqualObjects(event.stacktraceTypes, @[@"cocoa"]);
+    
+    error.type = BSGErrorTypeReactNativeJs;
+    XCTAssertEqualObjects(event.stacktraceTypes, @[@"reactnativejs"]);
+    
+    NSArray *(^ sorted)(NSArray *) = ^(NSArray *array) { return [array sortedArrayUsingSelector:@selector(compare:)]; };
+    
+    error.stacktrace = @[
+        [BugsnagStackframe frameFromJson:@{}],
+        [BugsnagStackframe frameFromJson:@{@"type": @"cocoa"}],
+        [BugsnagStackframe frameFromJson:@{@"type": @"reactnativejs"}],
+    ];
+    XCTAssertEqualObjects(sorted(event.stacktraceTypes), (@[@"cocoa", @"reactnativejs"]));
+    
+    event.errors = @[[[BugsnagError alloc] init]];
+    
+    BugsnagThread *thread1 = [[BugsnagThread alloc] init];
+    thread1.stacktrace = @[
+        [BugsnagStackframe frameFromJson:@{@"type": @"c"}],
+        [BugsnagStackframe frameFromJson:@{@"type": @"java"}],
+    ];
+    thread1.type = BSGThreadTypeCocoa;
+    event.threads = @[thread1];
+    XCTAssertEqualObjects(sorted(event.stacktraceTypes), (@[@"c", @"cocoa", @"java"]));
+
+    BugsnagThread *thread2 = [[BugsnagThread alloc] init];
+    thread2.stacktrace = @[
+        [BugsnagStackframe frameFromJson:@{@"type": @"csharp"}],
+        [BugsnagStackframe frameFromJson:@{@"type": @"android"}],
+    ];
+    event.threads = @[thread1, thread2];
+    XCTAssertEqualObjects(sorted(event.stacktraceTypes), (@[@"android", @"c", @"cocoa", @"csharp", @"java"]));
```

Comment:
Why is it expecting `android`?

Reply:
Having already used `cocoa` and `reactnativejs` in error.stacktrace, and `c` and `java` in thread 1, I needed some additional values to use in thread 2 to ensure they all get picked up correctly.

Any string could have been used in this test but I thought sticking to the prescribed values would be safest.